xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1288 @@
+    /// higher as the input it becomes asked about needed size before any
+    /// PostProcess call.
+    /// @param handle identification data for stream
+    /// @param mode_id The mode inside add-on which must be performed on call. Id
+    /// is set from add-on by iModeNumber on AE_DSP_MODE structure during
+    /// RegisterMode callback, and can be defined from add-on as a structure
+    /// pointer or anything else what is needed to find it.
+    /// @return The needed size of output array or 0 if no changes within it
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int PostProcessNeededSamplesize(const ADDON_HANDLE handle, unsigned int mode_id) { return 0; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// Returns the time in seconds that it will take
+    /// for the next added packet to be returned to KODI.
+    /// @param handle identification data for stream
+    /// @param mode_id The mode inside add-on which must be performed on call. Id
+    /// is set from add-on by iModeNumber on AE_DSP_MODE structure during
+    /// RegisterMode callback, and can be defined from add-on as a structure
+    /// pointer or anything else what is needed to find it.
+    /// @return the delay in seconds
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual float PostProcessGetDelay(const ADDON_HANDLE handle, unsigned int mode_id) { return 0.0f; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+
+    ///
+    /// @brief DSP post processing
+    /// On the post processing can be things performed with additional channel
+    /// upmix like 6.1 to 7.1
+    /// or frequency/volume corrections, speaker distance handling, equalizer... .
+    /// All DSP add-ons allowed to-do this.
+    /// @param handle identification data for stream
+    /// @param mode_id The mode inside add-on which must be performed on call. Id
+    /// is set from add-on by iModeNumber on AE_DSP_MODE structure during
+    /// RegisterMode callback, and can be defined from add-on as a structure
+    /// pointer or anything else what is needed to find it.
+    /// @param array_in Pointer to input data memory
+    /// @param array_out Pointer to output data memory
+    /// @param samples Amount of samples inside array_in
+    /// @return Amount of samples processed
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int PostProcess(const ADDON_HANDLE handle, unsigned int mode_id, float **array_in, float **array_out, unsigned int samples) { return 0; }
```

Comment:
There are you fully right, to allow a change of input stream was a bit bad thing in my mind.

Thought on begin with them to allow the input also as temporary array to use for math parts. But was a bit stupid from me :)